pkg/account/service.go

```diff
@@ -0,0 +1,186 @@
+package account
+
+import (
+	"time"
+
+	"github.com/Optum/dce/pkg/errors"
+	validation "github.com/go-ozzo/ozzo-validation"
+)
+
+// Writer put an item into the data store
+type Writer interface {
+	Write(i *Account, lastModifiedOn *int64) error
+}
+
+// Deleter Deletes an Account from the data store
+type Deleter interface {
+	Delete(i *Account) error
+}
+
+// SingleReader Reads Account information from the data store
+type SingleReader interface {
+	Get(ID string) (*Account, error)
+}
+
+// MultipleReader reads multiple accounts from the data store
+type MultipleReader interface {
+	List(query *Account) (*Accounts, error)
+}
+
+// Reader data Layer
+type Reader interface {
+	SingleReader
+	MultipleReader
+}
+
+// WriterDeleter data layer
+type WriterDeleter interface {
+	Writer
+	Deleter
+}
+
+// ReaderWriterDeleter includes Reader and Writer interfaces
+type ReaderWriterDeleter interface {
+	Reader
+	WriterDeleter
+}
+
+// Eventer for publishing events
+type Eventer interface {
+	Publish() error
+}
+
+// Manager manages all the actions against an account
+type Manager interface {
+	Setup(adminRole string) error
+}
+
+// Service is a type corresponding to a Account table record
+type Service struct {
```

Comment:
This is the new Account Service.  This separates the data of the Account and helps provide a generic Account Servicer.